metadata-ingestion/src/datahub/cli/specific/dataset_cli.py

```diff
@@ -111,3 +104,125 @@ def _get_existing_siblings(graph: DataHubGraph, urn: str) -> Set[str]:
         return set(existing.siblings)
     else:
         return set()
+
+
+@dataset.command(
+    name="file",
+)
+@click.option("--lintCheck", required=False, is_flag=True)
+@click.option("--lintFix", required=False, is_flag=True)
+@click.argument("file", type=click.Path(exists=True))
+@upgrade.check_upgrade
+@telemetry.with_telemetry()
+def file(lintcheck: bool, lintfix: bool, file: str) -> None:
+    """Operate on a Dataset file"""
+
+    if lintcheck or lintfix:
+        import tempfile
+        from pathlib import Path
+
+        # Create a temporary file in a secure way
+        # The file will be automatically deleted when the context manager exits
+        with tempfile.NamedTemporaryFile(suffix=".yml", delete=False) as temp:
```

Comment:
why is delete=False?

Reply:
The `delete=False` parameter prevents automatic deletion of the temporary file when the context manager closes. This is necessary because we need the file to persist after the `with` block to compare it with the original, display any differences, and potentially copy it back. We're still ensuring proper cleanup by manually deleting the file in the `finally` block.

Reply:
I don't get it - the finally block with the temp_path.unlink in inside the `with` block. Even with the current impl, the temp file does not last beyond the `with` block. If we set delete = True, we could remove the try...finally entirely